src/main/kotlin/no/nav/helse/spion/domene/varsling/repository/PostgresVarslingRepository.kt

```diff
@@ -0,0 +1,117 @@
+package no.nav.helse.spion.domene.varsling.repository
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.helse.spion.domene.varsling.Varsling
+import java.sql.Date
+import java.sql.ResultSet
+import java.sql.Timestamp
+import java.time.LocalDateTime
+import java.util.*
+import javax.sql.DataSource
+
+class PostgresVarslingRepository(private val ds: DataSource, private val mapper: ObjectMapper) : VarslingRepository {
+
+    private val tableName = "varsling"
+    private val insertStatement = "INSERT INTO $tableName(data, status, opprettet, virksomhetsNr, uuid, dato) VALUES(?::json, ?, ?, ?, ?::uuid, ?)"
+    private val updateStatement = "UPDATE $tableName SET status = ?, behandlet = ? WHERE uuid = ?"
+    private val nextStatement = "SELECT * FROM $tableName WHERE status=? ORDER BY opprettet ASC LIMIT ?"
+    private val countStatement = "SELECT count(*) FROM $tableName WHERE status = ?"
+    private val deleteStatement = "DELETE FROM $tableName WHERE uuid = ?"
+
+    fun mapToDto(varsling: Varsling): VarslingDto {
+        return VarslingDto(
+                uuid = varsling.uuid,
+                data = mapper.writeValueAsString(varsling.liste),
+                status = varsling.status,
+                opprettet = varsling.opprettet,
+                dato = varsling.dato,
+                virksomhetsNr = varsling.virksomhetsNr
+        )
+    }
+
+    fun mapToDto(res: ResultSet): VarslingDto {
+        return VarslingDto(
+                data = res.getString("data"),
+                uuid = res.getString("uuid"),
+                status = res.getInt("status"),
+                opprettet = res.getTimestamp("opprettet").toLocalDateTime(),
+                behandlet = res.getTimestamp("behandlet")?.toLocalDateTime(),
+                dato = res.getDate("dato").toLocalDate(),
+                virksomhetsNr = res.getString("virksomhetsNr")
+        )
+    }
+
+    fun mapToDomain(dto: VarslingDto): Varsling {
+        return Varsling(
+                dato = dto.dato,
+                virksomhetsNr = dto.virksomhetsNr,
+                uuid = dto.uuid,
+                opprettet = dto.opprettet,
+                status = dto.status,
+                liste = mapper.readValue(dto.data)
+        )
+    }
+
+    override fun finnNesteUbehandlet(): Varsling {
+        return ds.connection.use {
+            val resultList = ArrayList<Varsling>()
+            val res = it.prepareStatement(nextStatement).apply {
```

Comment:
Her tenker jeg dato må være med. Vi vil hente varsler for en spesifikk dag. Ved hver kjøring er det for gårsdagen som gjelder. Sånn jeg ser det for meg i alle fall